thirdparty/patchmatchnet/datasets/data_io.py

```diff
@@ -0,0 +1,93 @@
+"""MVSDataset io utils
+    reference: https://github.com/FangjinhuaWang/PatchmatchNet
+
+"""
+import sys
+from typing import Tuple
+
+import re
+import numpy as np
+
+
+def read_pfm(filename: str) -> Tuple[np.ndarray, float]:
+    """Read a depth map from a .pfm file
+
+    Args:
+        filename: .pfm file path string
+
+    Returns:
+        data: array of shape (H, w, C) representing loaded heat map
+        scale: float to recover heat map pixel values to actual probabilities
+    """
+    file = open(filename, "rb")
+    color = None
+    width = None
+    height = None
+    scale = None
+    endian = None
+
+    header = file.readline().decode("utf-8").rstrip()
+    if header == "PF":
+        color = True
+    elif header == "Pf":
+        color = False
+    else:
+        raise Exception("Not a PFM file.")
+
+    dim_match = re.match(r"^(\d+)\s(\d+)\s$", file.readline().decode("utf-8"))
+    if dim_match:
+        width, height = map(int, dim_match.groups())
+    else:
+        raise Exception("Malformed PFM header.")
+
+    scale = float(file.readline().rstrip())
+    if scale < 0:  # little-endian
+        endian = "<"
+        scale = -scale
+    else:
+        endian = ">"  # big-endian
+
+    data = np.fromfile(file, endian + "f")
+    shape = (height, width, 3) if color else (height, width, 1)
+
+    data = np.reshape(data, shape)
+    data = np.flipud(data)
+    file.close()
+    return data, scale
+
+
+def save_pfm(filename: str, image: np.ndarray, scale: float = 1) -> None:
+    """Save a depth map from a .pfm file
+
+    Args:
+        filename: output .pfm file path string,
+        image: depth map to output,
+        scale: scale parameter to output
```

Comment:
maybe `scale: scale parameter to save` is clearer. Maybe I would also clarify that the depth map passed into `save_pfm()` already exists at this scale?